resteasy/src/main/java/com/cerner/beadledom/resteasy/ModuleProcessor.java

```diff
@@ -0,0 +1,80 @@
+package com.cerner.beadledom.resteasy;
+
+import com.google.inject.Binding;
+import com.google.inject.Injector;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.plugins.guice.GuiceResourceFactory;
+import org.jboss.resteasy.plugins.guice.i18n.LogMessages;
+import org.jboss.resteasy.plugins.guice.i18n.Messages;
+import org.jboss.resteasy.spi.Registry;
+import org.jboss.resteasy.spi.ResourceFactory;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.jboss.resteasy.util.GetRestful;
+
+/**
+ * Processes a given life cycle injector that contains Guice modules.
+ *
+ * <p>This class expands upon {@link org.jboss.resteasy.plugins.guice.ModuleProcessor} by enforcing
+ * restrictions on available JAX-RS {@link Provider}s to be added to {@link ResteasyProviderFactory}.
+ *
+ * @author Charan Panchagnula
+ */
+public class ModuleProcessor {
+
+  private final Registry registry;
+  private final ResteasyProviderFactory providerFactory;
+
+  /**
+   * Constructor for ModuleProcessor.
+   *
+   * @param registry {@link Registry} for adding resource implementation endpoints
+   * @param providerFactory {@link ResteasyProviderFactory} used for registering JAX-RS {@link Provider} instances
+   */
+  public ModuleProcessor(final Registry registry, final ResteasyProviderFactory providerFactory) {
+    Objects.requireNonNull(registry, "registry: null");
```

Comment:
What exception does this throw if registry is null? Is it a NullPointer? Would it make more sense as an ArgumentException?

Reply:
Using IllegalArgumentException vs NullPointerException is a holy war i don't want to start now. If i want to use IAE, i'll have to use Preconditions.checkArgument which isn't a native utility so i stuck with Objects, which is also something used throughout beadledom